src/common/transformations/src/transformations/common_optimizations/matmul_split_decomposition.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/matmul_split_decomposition.hpp"
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace ov;
+using namespace ov::pass::pattern;
+
+void pass::MatmulGatherDecomposition::split_weights(const Output<Node>& weights,
+                                                    OutputVector& new_weights,
+                                                    Output<Node>* bias,
+                                                    OutputVector& new_bias,
+                                                    const bool transpos_b) {
+    // weights is static
+    if (weights.get_partial_shape().size() != 2u) {
+        return;
+    }
+
+    if (bias) {
+        const auto& bias_shape = bias->get_partial_shape();
+        if (bias_shape.is_dynamic()) {
+            return;
+        }
+        auto bias_rank = bias_shape.rank().get_length();
+        if (bias_rank != 3 && bias_rank != 1) {
+            return;
+        }
+    }
+
+    // Decompose weights
+    auto axis = register_new_node(op::v0::Constant::create(element::i32, Shape{}, {transpos_b ? 0 : 1}));
+    auto split = register_new_node<op::v1::Split>(weights, axis, 3);
```

Comment:
Can we make a magic number `3` some constexpr variable (it can be a transformation class member, for instance)? This would allow us to easily extend the transformation on other cases if needed

Reply:
Updated